webknossos-datastore/app/com/scalableminds/webknossos/datastore/rpc/RPCRequest.scala

```diff
@@ -215,79 +232,53 @@ class RPCRequest(val id: Int, val url: String, wsClient: WSClient)(implicit ec:
 
   private def extractBytesResponse(r: Fox[WSResponse]): Fox[Array[Byte]] =
     r.flatMap { response =>
-      if (Status.isSuccessful(response.status)) {
-        val responseBytes = response.bodyAsBytes
-        if (verbose) {
-          logger.debug(
-            s"Successful request (ID: $id). " +
-              s"Body: '<${responseBytes.length} raw bytes>'")
-        }
-        Fox.successful(responseBytes.toArray)
-      } else {
-        logger.error(
-          s"Failed to send WS request to $url (ID: $id). " +
-            s"RequestBody: '$requestBodyPreview'. Status ${response.status}. " +
-            s"ResponseBody: '${response.body.take(100)}'")
-        Fox.failure("Unsuccessful WS request to $url (ID: $id)")
```

Comment:
Is there a reason why you removed this error handling here and below? It is because this case cannot happen anyway, because only successful requests fill end up in the flatMap method body? If that's true, why not keep the error message by some kind of  `flatMap(...).orElse(//handle error here)` manor. Don't know how well this can be applied here :thinking:

Reply:
Oh nvm. Perform request does this. :+1: